### PR TITLE
(WIP) Add users-permissions updateMe

### PR DESCRIPTION
**Pull request by derrickmehaffy**
Signed-off-by: Derrick Mehaffy <derrickmehaffy@gmail.com>

### What does it do?

Adds a PUT /users/me route by default to the users-permissions plugin

### Why is it needed?

Requested by users

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/2818

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/controllers/User.js [hunk 1]
```diff
@@ -122,4 +123,22 @@ module.exports = {
...
+  async updateMe(ctx) {
+    const input = ctx.request.body;
+
+    try {
+      await validateProfileUpdateInput(input);
+    } catch (err) {
+      return ctx.badRequest('ValidationError', err);
+    }
+
+    const updatedUser = await strapi.plugins['users-permissions'].services.user.edit(ctx.state.user.id, input);
```
Comment (RajNandigalla) [inline packages/strapi-plugin-users-permissions/controllers/User.js:140]: Incase some one needs this 
Change - `services.user.edit({ id: ctx.state.user.id }, input);`

Comment (derrickmehaffy) [reply packages/strapi-plugin-users-permissions/controllers/User.js:140]: I wouldn't recommend using this yet, it's only a prototype with no validation or security yet.

Comment (RajNandigalla) [reply packages/strapi-plugin-users-permissions/controllers/User.js:140]: Sure
This Pointed me in the right direction , i have added my own validations

**Comment by derrickmehaffy** [comment]
This pull request has been mentioned on **Strapi Community Forum**. There might be relevant details there:

https://forum.strapi.io/t/applying-isowner-policy-to-user-permissions-plugin/1782/2

**Comment by Mo9a7i** [comment]
What's the plan for this one ?

**Comment by derrickmehaffy** [comment]
> What's the plan for this one ?

Currently nothing, I'm marking as closed as I don't have time to finish it. Someone else can pick this up if they like.

**Pull request closed without merging**